module/zfs/zpl_file.c

```diff
@@ -361,8 +361,10 @@ zpl_write(struct file *filp, const char __user *buf, size_t len, loff_t *ppos)
 	ssize_t wrote;
 
 	crhold(cr);
+	spl_inode_lock(file_inode(filp));
 	wrote = zpl_write_common(filp->f_mapping->host, buf, len, ppos,
```

Comment:
@tuxoko @behlendorf Is there any reason we are getting the inode from f_mapping->host and not directly from file_inode? That'd be one dereference and make the code more succinct and clear. In my testing I put an assert for f_inode == f_mapping_host and it never fired, so I assume it's safe?